rideshare.rb

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/ruby
+# 
+# Title  : Project Ride Share - Ada Cohort 13
+# Author : Suely Barreto
+# Date   : February 2020
+
+##########################################################
+# Step 1: Establish the layers
+# In this section of the file, as a series of comments,
+# create a list of the layers you identify.
+# Which layers are nested in each other?
+# Which layers of data "have" within it a different layer?
+# Which layers are "next" to each other?
+########################################################## 
+
+## Solution
+    ## There is 1 CSV file with ride information (11 lines and 5 columns)
+    ## Each line has information for 1 ride (driver_id, ride_date, ride_cost, rider_id, ride_rating)
+    ## You can summarize driver information for the 4 drivers (rides, earnings, and ratings)
+
+########################################################
+# Step 2: Assign a data structure to each layer
+# Copy your list from above, and in this section
+# determine what data structure each layer should have
+########################################################
+
+## Solution
+  ## 1 array (rides) with 11 hashes (ride), each with 5 keys (driver_id, ride_date, ride_cost, rider_id, ride_rating)
+  ## 1 hash (drivers) with a string key (driver) with a value that is a hash (driver_info) containing rides, earnings, ratings
+
+########################################################
+# Step 3: Make the data structure!
+# Setup the entire data structure:
+########################################################
+
+# Function to get the average rating
+def get_average(total_rating, rides)
+
+  if rides > 0
+    average = total_rating.to_f / rides
+  else
+    average = 0
+  end
+  return average
+
+end
+
+# Create an array from the CSV file
+require "csv"
+rides = []
+CSV.open("rides.csv", 'r', headers: true).each do |ride_array|
+  ride_hash = Hash.new
+  ride_hash[:driver_id]   = ride_array[0]
+  ride_hash[:ride_date]   = ride_array[1]
+  ride_hash[:ride_cost]   = ride_array[2].to_f
+  ride_hash[:rider_id]    = ride_array[3]
+  ride_hash[:ride_rating] = ride_array[4].to_f
+  rides << ride_hash
+end
+
+# Create a hash for drivers with data from the rides
+drivers = Hash.new
```

Comment:
In ruby we usually would write this as `drivers = {}`. `Hash.new` is a syntax we generally reserve for other data types.